src/main/java/org/apache/maven/plugins/clean/CleanMojo.java

```diff
@@ -190,7 +193,25 @@ public void execute()
             return;
         }
 
-        Cleaner cleaner = new Cleaner( session, getLog(), isVerbose(), fast ? fastDir : null );
+        File fastDir;
+        if ( fast && this.fastDir != null )
+        {
+            fastDir = this.fastDir;
+        }
+        else if ( fast && multiModuleProjectDirectory != null )
+        {
+            fastDir = new File( multiModuleProjectDirectory, "target/.clean" );
```

Comment:
I wonder what will happen when `build.outputDirectory` has been modified...

Reply:
Nothing good, but given this is an opt-in feature, people can define the `fastDir` directory explicitely if needed.
I'm writing the docs on those new properties, so I'll add some info.

Reply:
Agreed.